test/compress/reduce_vars.js

```diff
@@ -4544,3 +4544,42 @@ issue_2455: {
         }
     }
 }
+
+issue_2560: {
+    options = {
+        reduce_funcs: true,
+        reduce_vars: true,
+        toplevel: true,
+        unused: true,
+    }
+    input: {
+        function main(state) {
+            var thing = baz(state);
+            if (thing !== (thing = baz(state)))
```

Comment:
I think `state` is unused in all cases and can be removed.

Reply:
Indeed - the resulting test would still fail on `master`.

Done in https://github.com/mishoo/UglifyJS2/pull/2563/commits/368209ebb1bbf7beb68897309086034b1e7f1712